llvm/lib/Target/AIE/AIEBundle.h

```diff
@@ -103,6 +105,16 @@ template <class I> class Bundle {
       MetaInstrs.push_back(Instr);
       return;
     }
+
+    // Keep track of BUNDLE instructions. They need to be cleaned up when
+    // de-bundling before re-bundling. See applyBundles()
+    if (Instr->getOpcode() == TargetOpcode::BUNDLE) {
```

Comment:
super nit: `Instr->isBundle()`

Reply:
Oh I remember now, the reason I chose `getOpcode() == ...` is because `AIE::Bundle` is a template class that should work with both `MachineInstr` and `MCInstr`. The latter does not provide `::isBundle()`